### ISSUE TITLE
UX: Different strategies of toolbar positioning

### ISSUE THREAD

**Issue by oleq** [ckeditor/ckeditor5-design#96]
# How to make the toolbar useful but not obtrusive?
- Can toolbar follow the caret and not block the access to the content?
- When the toolbar can be sticky to the viewport and not impede the navigation of the webpage? Is this possible? (like CKE4 notifications)
  - **Note:** If the toolbar is sticky to the viewport, then it can be shared between editor instances.
  - Would [`position:sticky`](https://developer.mozilla.org/en-US/docs/Web/CSS/position#Sticky_positioning) be a solution?
- How to save space in the viewport without hiding (or pushing away) the toolbar?

---
## Sticky to the top of editable

![image](https://cloud.githubusercontent.com/assets/1099479/12417431/5bab348a-bea8-11e5-880a-842fd0a9506a.png)
## Following the caret

![image](https://cloud.githubusercontent.com/assets/1099479/12417432/60bdbb5a-bea8-11e5-98f2-3ec3d4092dc3.png)
## Always sticky to the viewport

![image](https://cloud.githubusercontent.com/assets/1099479/12417436/667d7e72-bea8-11e5-9be2-d37728241fb3.png)
## CSS position:sticky

![image](https://cloud.githubusercontent.com/assets/1099479/12417439/6c676000-bea8-11e5-9503-56902d6ecbad.png)

**Comment by wimleers** [comment]
I can just say one thing: following around, while you scroll through content, is _supremely_ annoying and causes significant problems when wanting to select content _under_ the toolbar. This is a problem Aloha Editor has (or at least used to have) and that Drupal 8's in-place editing toolbar still has.

**Comment by fredck** [comment]
Aloha was used to do that with a toolbar close to the selection and this is indeed extremely annoying.

I still like the way we handle this in CKEditor 4. The [demo](http://ckeditor.com/demo#inline) can be used to check it. Just pick one of the longer editables using a small browser window and you can check the behavior of it.

**Comment by oleq** [comment]
I also like the approach that we use in CKEditor 4. But it could be that I got so used to it that I'm blind to other solutions. This is why I think we need more feedback from "the outside".

**Comment by ofcapl** [comment]
This:

> I can just say one thing: following around, while you scroll through content, is supremely annoying and causes significant problems when wanting to select content under the toolbar. 

+1

> Would position:sticky be a solution?

I don't know how wide range of browsers do You want to support, but take a look at this `position:sticky` support table: http://caniuse.com/#feat=css-sticky

**Comment by fredck** [comment]
@ofcapl, I think we're talking here about simulating the behavior of position:sticky among all browsers. We do this already in CKEditor 4.

**Comment by ofcapl** [comment]
@fredck thanks for the info - edited my previous comment :)

**Comment by oleq** [comment]
> I can just say one thing: following around, while you scroll through content, is supremely annoying and causes significant problems when wanting to select content under the toolbar. This is a problem Aloha Editor has (or at least used to have) and that Drupal 8's in-place editing toolbar still has.

@wimleers I checked D8 and noticed that in–place editing toolbar is either attached to the top (arrow down) or bottom (arrow up) edge of the content. It never floats over the content. Could you extend your comment (some screenshot)? Or maybe I completely missed your point.

**Comment by koleary** [comment]
The problem that the in-place editing toolbar in Drupal 8 has is that it "floats", that is, it shifts position relative to the text-area underneath it. This is a usability bug we need to fix in Drupal because the toolbar will sometimes cover text that you want to edit and the user needs to scroll, or click elsewhere or actually close the toolbar sometimes to reveal the text.

We introduced the "floating behavior" animation to make the experience smoother and to handle scroll (the bar used CSS position sticky). However the imprecision of the animation and the size of the bar have caused problems as I said before.

In the time since we first introduced it I have come around to the view that the bar needs to be divided into two separate widgets (as it is in Medium) one for adding things (images, tables, media, etc.), and the other solely for formatting. Where the "add" bar goes by default is a question that is handled in a variety of ways but a formatting bar that appears directly at the cursor when text is selected is becoming the accepted standard in many contexts, from iOS to Medium to Squarespace and IMO that's where CK should go as well.
